tools/alerts/alert_utils.py

```diff
@@ -0,0 +1,168 @@
+      nodes {
+        id
+        title
+        closed
+        number
+        body
+        createdAt
+        comments(first: 100) {
+          nodes {
+            bodyText
+            databaseId
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+NUM_ISSUES_QUERY = """
+query ($query: String!) {
+  search(type: ISSUE, query: $query) {
+    issueCount
+  }
+}
+"""
+
+REPO_OWNER = "pytorch"
+PYTORCH_REPO_NAME = "pytorch"
+TEST_INFRA_REPO_NAME = "test-infra"
+PYTORCH_ALERT_LABEL = "pytorch-alert"
+FLAKY_TESTS_LABEL = "module: flaky-tests"
+NO_FLAKY_TESTS_LABEL = "no-flaky-tests-alert"
+FLAKY_TESTS_SEARCH_PERIOD_DAYS = 14
+DISABLED_ALERTS = [
+    "rerun_disabled_tests",
+    "unstable",
+]
+
+headers = {"Authorization": f"token {os.environ.get('GITHUB_TOKEN')}"}
+CREATE_ISSUE_URL = (
+    f"https://api.github.com/repos/{REPO_OWNER}/{TEST_INFRA_REPO_NAME}/issues"
+)
+UPDATE_ISSUE_URL = (
+    f"https://api.github.com/repos/{REPO_OWNER}/{TEST_INFRA_REPO_NAME}/issues/"
+)
+
+GRAPHQL_URL = "https://api.github.com/graphql"
+
+# rename this when these are ready
+# PYTORCH_ALERT_LABEL = "pytorch-alert"
+PYTORCH_ALERT_LABEL = "pytorch-alert-test"
+
+headers = {"Authorization": f"token {os.environ.get('GITHUB_TOKEN')}"}
+
+def fetch_alerts(
+    labels: List[str],
+    alert_repo_owner: str = REPO_OWNER,
+    alert_repo_name: str = TEST_INFRA_REPO_NAME,
+) -> List[Any]:
+    try:
+        variables = {
+            "owner": alert_repo_owner,
+            "name": alert_repo_name,
+            "labels": labels,
+        }
+        r = requests.post(
+            GRAPHQL_URL,
+            json={"query": ISSUES_WITH_LABEL_QUERY, "variables": variables},
+            headers=headers,
+        )
+        r.raise_for_status()
+        return json.loads(r.text)["data"]["repository"]["issues"]["nodes"]
+    except Exception as e:
+        raise RuntimeError("Error fetching alerts", e)
+
+ 
+
+def fetch_alerts_filter(repo: str, labels: List[str], alertType: str) -> List[Any]:
+    alerts = fetch_alerts(labels)
+    return [
+        alert
+        for alert in alerts
+        if f"{repo}" in alert["title"] and alertType in alert["title"]
+    ]
+
+def _assert_same_repo_and_type(alerts: List[Dict[str, Any]]) -> None:
+    repo = alerts[0]["repo"]
+    alert_type = alerts[0]["AlertType"]
+    alert_org = alerts[0]["org"]
+    for alert in alerts:
+        if alert["repo"] != repo:
+            raise ValueError(
+                f"Alerts must be from the same repository, got {repo} and {alert['repo']}"
+            )
+        if alert["AlertType"] != alert_type:
+            raise ValueError(
+                f"Alerts must be of the same type, got {alert_type} and {alert['AlertType']}"
+            )
+        if alert["org"] != alert_org:
+            raise ValueError(
+                f"Alerts must be of the same org, got {alert_org} and {alert['org']}"
+            )
+
+def update_issue(
+    issue: Dict, old_issue: Any, dry_run: bool = False
```

Comment:
It's interesting to see that the existing alert script already has dry run support, so I think this is a feature that we should keep